src/com/epam/helloworld/greenhouse/MyProgram.java

```diff
@@ -2,14 +2,23 @@
 
 public class MyProgram {
     public static void main(String[] args) {
-        Plants plant1 = PlantFactory.createPlant("floweringPlant", "African Violet", "Tanzania", "Needs bright indirect light", "Flowering Plant");
-        Plants plant2 = PlantFactory.createPlant("shrub", "Boxwood", "Europe", "Needs well-draining soil", "Shrub");
-        Plants plant3 = PlantFactory.createPlant("housePlant", "Peace Lily", "Central and South America", "Needs moderate watering", "House Plant");
-        Plants plant4 = PlantFactory.createPlant("housePlant", "Pothos", "Solomon Islands", "Needs occasional misting", "House Plant");
-        Plants plant5 = PlantFactory.createPlant("shrub", "Lilac", "Europe", "Needs full sun exposure", "Shrub");
-        Plants plant6 = PlantFactory.createPlant("floweringPlant", "Sunflower", "North America", "Requires well-drained soil", "Flowering Plant");
-        Plants plant7 = PlantFactory.createPlant("housePlant", "Snake Plant", "West Africa", "Requires low light", "House Plant");
-        Plants plant8 = PlantFactory.createPlant("shrub", "Azalea", "Asia", "Prefers acidic soil", "Shrub");
+
+        //Objects of the inner class
+        HousePlant.HouseplantCare plant3Care = new HousePlant("Peace Lily", "Central and South America", "Needs moderate watering", "House Plant", null).new HouseplantCare("Water twice a week", true, "Well-draining potting mix");
```

Comment:
code duplication. Fields for HousePlant filling in here and again on the line 13